docs/import/update_firestore_disaster.js

```diff
@@ -0,0 +1,57 @@
+import {getFirestoreRoot} from '../firestore_document.js';
+import {getDisaster} from '../resources.js';
+
+export {updateDataInFirestore};
+
+const STATE = {
+  SAVED: 0,
+  WRITING: 1,
+  QUEUED_WRITE: 2,
+};
+Object.freeze(STATE);
+
+let state = STATE.SAVED;
+let pendingWriteCount = 0;
+
+window.onbeforeunload = () => pendingWriteCount > 0 ? true : null;
+
+/**
+ * Writes the current state of a disaster's data to firestore.
+ * @param {Function} dataSupplier Function that returns data to be written for
+ *     current disaster
+ * @param {Function} startCallback Function that should be called before a write
+ *     starts (to indicate loading progress, for instance)
+ * @param {Function} finishCallback Function called whenever a write ends
+ * @return {?Promise<void>} Returns when finished writing or null if it just
+ * queued a write and doesn't know when that will finish.
+ */
+function updateDataInFirestore(dataSupplier, startCallback, finishCallback) {
```

Comment:
maybe make startCallback and finishCallback optional?

Reply:
My suspicion is that they won't be, because we'll need to know when all writes have finished for deletions. So I think not worth it.